zio-http/src/main/scala/zhttp/http/HttpData.scala

```diff
@@ -54,7 +54,7 @@ sealed trait HttpData[-R, +E] { self =>
    */
   private[zhttp] def unsafeSize: Long = self match {
     case HttpData.Empty               => 0L
-    case HttpData.Text(text, _)       => text.length.toLong
+    case HttpData.Text(text, charSet) => Unpooled.copiedBuffer(text, charSet).readableBytes().toLong
```

Comment:
this is a very expensive way to compute size.

Reply:
Agree creating new buffer is expensive. I will re-look.

Reply:
Done.